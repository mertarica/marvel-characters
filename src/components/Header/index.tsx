import React from "react";
import Container from "react-bootstrap/Container";
import Navbar from "react-bootstrap/Navbar";

import Logo from "../../assets/images/path-logo.svg";

const Header = () => {
  return (
    <Navbar className="c-header" data-testid="header">
      <Container>
        <Navbar.Brand href="/">
          <img src={Logo} alt="Path Logo" />
        </Navbar.Brand>
        <Navbar.Toggle />
        <Navbar.Collapse className="justify-content-end">
          <Navbar.Text className="c-header__text">
            <strong>MARVEL</strong>
            <span className="c-header__text--light">Heroes</span>
          </Navbar.Text>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Header;
