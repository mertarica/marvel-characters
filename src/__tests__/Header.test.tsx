import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Header from "../components/Header";

describe("<Header />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    render(<Header />);
    expect(screen.getByTestId("header")).toBeInTheDocument();
  });

  it("renders logo", () => {
    render(<Header />);
    const imgElement = screen.getByAltText(/logo/i);
    expect(imgElement).toBeInTheDocument();
  });
});
