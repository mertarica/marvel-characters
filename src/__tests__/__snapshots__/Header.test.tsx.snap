// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<nav
  className="c-header navbar navbar-expand navbar-light"
  data-testid="header"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Path Logo"
        src="path-logo.svg"
      />
    </a>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <span
        className="c-header__text navbar-text"
      >
        <strong>
          MARVEL
        </strong>
        <span
          className="c-header__text--light"
        >
          Heroes
        </span>
      </span>
    </div>
  </div>
</nav>
`;
